control-plane/csi-driver/src/bin/node/filesystem_vol.rs

```diff
@@ -324,6 +326,14 @@ pub(crate) fn unpublish_fs_volume(msg: &NodeUnpublishVolumeRequest) -> Result<()
         // The idempotency requirement means this is not an error.
         // Just clean up as best we can and claim success.
 
+        // If the Device or Resource busy error is encountered, return error so that 
+        // unmounting and removing is retried 
+        if error.to_string().contains(RESOURCE_BUSY_ERROR_STRING){
```

Comment:
Checking error by parsing error string here, because ErrorKind::ResourceBusy is an unstable rust feature. 
https://doc.rust-lang.org/stable/std/io/enum.ErrorKind.html#variant.ResourceBusy

Reply:
I may be looking at this wrongly in the gh website, but is this check if the right place? I can't see the variable error in the current scope?
That aside, I wonder if we should switch the logic, and NotFound becomes the only "accepted" error, and everything else should lead into a failure, what do you think?

Reply:
Yep, that's what I was thinking too, but I wasn't sure why we return success in other cases currently. Sure, I would make the change to return error in all other cases.